lib/shopify-cli/commands.rb

```diff
@@ -6,10 +6,16 @@ module Commands
       default: 'help',
       contextual_resolver: nil,
     )
+    @core_commands = []
 
     def self.register(const, cmd, path = nil)
       autoload(const, path) if path
       Registry.add(->() { const_get(const) }, cmd)
+      @core_commands.push(cmd)
+    end
+
+    def self.core_command?(cmd)
```

Comment:
Perhaps this could just be `core?` since we already say 'command' in the class name and argument?

Reply:
Maybe `reserved?` instead of `core?` or `core_command?`

Reply:
Hmm, I'd be a little hesitant to introduce another concept alongside `core` and `global`. All `core` commands would now be implicitly reserved already, right?